File: Tune/app/src/main/java/com/example/tune/activities/adapters/CustomTimelineAdapter.java
package com.example.tune.activities.adapters;

import android.app.Activity;
import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.core.content.ContextCompat;

import com.example.tune.activities.PostActivity;
import com.example.tune.R;

import com.example.tune.activities.models.Post;

public class CustomTimelineAdapter extends ArrayAdapter {

    private Post[] posts;
    private Activity context;

    public CustomTimelineAdapter(Activity context, Post[] posts) {
        super(context, R.layout.timeline_row, posts);
        this.posts = posts;
        this.context = context;
    }

    public View getView(final int position, View view, ViewGroup parent){
        LayoutInflater inflater = context.getLayoutInflater();
        View rowView = inflater.inflate(R.layout.timeline_row, null, true);

        TextView userTxt = (TextView) rowView.findViewById(R.id.timelineUser);
        TextView genreTxt = (TextView) rowView.findViewById(R.id.timelineGenre);
        TextView locationTxt = (TextView) rowView.findViewById(R.id.timelineLocation);

        userTxt.setText(posts[position].getUser().getName());
        genreTxt.setText(posts[position].getGenre().toString());
        locationTxt.setText(posts[position].getLocation());

        LinearLayout linearLayout = rowView.findViewById(R.id.linearLayoutOnClick);
        linearLayout.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(context, PostActivity.class);
                intent.putExtra("POST_INFORMATION", posts[position]);
                context.startActivity(intent);
            }
        });

        if(position % 2 == 1){
            rowView.setBackgroundColor(ContextCompat.getColor(context, R.color.lightBlue));
        } else {
            rowView.setBackgroundColor(ContextCompat.getColor(context, R.color.darkBlue));
        }

        return rowView;
    }

}

File: tune_kotlin/app/src/main/java/com/example/tune_kotlin/adapters/CustomTimelineAdapter.kt
package com.example.tune_kotlin.adapters

import android.content.Context
import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.LinearLayout
import android.widget.TextView
import androidx.core.content.ContextCompat
import com.example.tune_kotlin.R
import com.example.tune_kotlin.activities.PostActivity
//import com.example.tune_kotlin.activities.PostActivity
import com.example.tune_kotlin.models.Post

class CustomTimelineAdapter(private val context: Context, private val posts: List<Post>) :BaseAdapter() {
    override fun getItem(position: Int): Any {
        return posts[position]
    }

    override fun getItemId(position: Int): Long {
        return posts[position].id ?:0

    }

    override fun getCount(): Int {
        return posts.size
    }

    override fun getView(position: Int, view: View?, parent: ViewGroup): View {
        val inflater: LayoutInflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
        val rowView = inflater.inflate(R.layout.timeline_row, null, true)
        if(posts.size != 0) {
            val userTxt = rowView.findViewById(R.id.timelineUser) as TextView
            val genreTxt = rowView.findViewById(R.id.timelineGenre) as TextView
            val locationTxt = rowView.findViewById(R.id.timelineLocation) as TextView

            userTxt.text = posts[position].email
            genreTxt.text = posts[position].genre?.name
            locationTxt.text = posts[position].location


            if (position % 2 == 1) {
                rowView.setBackgroundColor(ContextCompat.getColor(context, R.color.lightBlue))
            } else {
                rowView.setBackgroundColor(ContextCompat.getColor(context, R.color.darkBlue))
            }
        } else {
            val userTxt = rowView.findViewById(R.id.timelineUser) as TextView
            userTxt.text = "No posts available! Check your preferences."

        }

        rowView.setOnClickListener{
            val intent = Intent(context, PostActivity::class.java)
            intent.putExtra("POST_INFORMATION", posts[position])
            context.startActivity(intent)
        }

        return rowView
    }

}
